cmd/provider/internal/config/identity_test.go

```diff
@@ -0,0 +1,31 @@
+package config
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestDecodeOrCreate(t *testing.T) {
+	// first, creating an identity and saving the key to the file
+	identity := Identity{}
+	privKeyPath := filepath.Join(t.TempDir(), "privKey")
+	t.Setenv(PrivateKeyPathEnvVar, privKeyPath)
+
+	peerIDBefore, pkBefore, err := identity.DecodeOrCreate(io.Discard)
+	require.NoError(t, err)
+
+	_, err = os.Stat(privKeyPath)
+	require.NoError(t, err)
```

Comment:
Assert that the identity is written out onto the file path and once decoded its ID matches the ID returned by `DecodeOrCreate`.